packages/frontend-2/pages/server-management/index.vue

```diff
@@ -0,0 +1,210 @@
+<template>
+  <div>
+    <Portal to="navigation">
+      <HeaderNavLink
+        :to="'/server-management'"
+        name="Server Management"
+      ></HeaderNavLink>
+    </Portal>
+    <div
+      class="flex flex-col md:flex-row space-y-2 space-x-2 justify-between mb-4 md:items-center"
+    >
+      <div>
+        <h5 class="h4 font-bold">Your server at a glance</h5>
+      </div>
+    </div>
+
+    <div class="grid grid-cols-1 md:grid-cols-2 gap-6 mt-10">
+      <Card :server-info="serverData" @cta-clicked="showDialog = true" />
+
+      <SettingsDialog
+        ref="settingsDialog"
```

Comment:
Very rarely if ever you would need to use `refs` and the ability to invoke some internal component methods. In this case it does make sense, cause there's no other way to trigger submission, but I don't see the need to have buttons be injectable from outside anyway. You already know what the buttons are going to be inside the `SettingDialog.vue` component, so why not define them there and thus remove the need to expose submission outwards?